Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -780,7 +780,13 @@ class BrowserViewController: UIViewController {
         }
         
         func shouldShowTabBar() -> Bool {
-            let tabCount = tabManager.tabsForCurrentMode.count
+            let tabCount: Int
+            if !PrivateBrowsingManager.shared.isPrivateBrowsing && tabManager.selectedTab?.isPrivate == true {
```

Comment:
Why is `tabsForCurrentMode` not returning the correct data? Is this happening too soon? Then we should probably find a way to fix the timing. This feels pretty hacky.

Reply:
Changed the implementation to setting the mode at source.